### PR TITLE
Fix Controllers and add unit tests

### PR DESCRIPTION
**Pull request by avidraccoon**
This pull request introduces several improvements and refactorings to the controller handling system, focusing on configuration flexibility, code maintainability, and comprehensive testing. The main changes include standardizing configuration field names, refactoring how default value bounds are set for controller elements, and adding a suite of unit tests and supporting infrastructure to ensure robust JSON-based controller configuration loading and simulation.

**Controller configuration and code refactoring:**

* Standardized the JSON field name for the low-level control element from `"controllerInterface"` to `"humanControlElement"` throughout the codebase and documentation, ensuring consistency and clarity in configuration files and code comments. [[1]](diffhunk://#diff-8f02e24df5e2ff512f9a6bab6ac4e0be652512f94f40948c4b202eb4378cecceL921-R921) [[2]](diffhunk://#diff-8f02e24df5e2ff512f9a6bab6ac4e0be652512f94f40948c4b202eb4378cecceL961-R961) [[3]](diffhunk://#diff-8f02e24df5e2ff512f9a6bab6ac4e0be652512f94f40948c4b202eb4378cecceL978-R978)
* Refactored the initialization of `minValue` and `maxValue` for `Button`, `Axis`, and `POV` controller elements: removed default-setting constructors and instead set these defaults in the `initialize` method, allowing for dynamic assignment based on the associated low-level control element if present. [[1]](diffhunk://#diff-8f02e24df5e2ff512f9a6bab6ac4e0be652512f94f40948c4b202eb4378cecceL1372-L1387) [[2]](diffhunk://#diff-8f02e24df5e2ff512f9a6bab6ac4e0be652512f94f40948c4b202eb4378cecceR1453-R1456) [[3]](diffhunk://#diff-8f02e24df5e2ff512f9a6bab6ac4e0be652512f94f40948c4b202eb4378cecceL1488-L1498) [[4]](diffhunk://#diff-8f02e24df5e2ff512f9a6bab6ac4e0be652512f94f40948c4b202eb4378cecceR1493-R1501) [[5]](diffhunk://#diff-8f02e24df5e2ff512f9a6bab6ac4e0be652512f94f40948c4b202eb4378cecceL1534-L1545) [[6]](diffhunk://#diff-8f02e24df5e2ff512f9a6bab6ac4e0be652512f94f40948c4b202eb4378cecceR1537-R1546)
* Changed the default deadband threshold for `Button` inputs from `0.0` to `0.5` to better filter out noise and unintentional input.

**Improvements to controller JSON representation:**

* Made `ControllerJsonRepresentation` constructor public and added a static method to retrieve its constructor via reflection, improving integration with JSON conversion utilities. [[1]](diffhunk://#diff-0d2cc65f133b6374a202d7521bdc4659749ec5381ddd5c4075d0046b7b7fea97L29-R30) [[2]](diffhunk://#diff-0d2cc65f133b6374a202d7521bdc4659749ec5381ddd5c4075d0046b7b7fea97R81-R85)
* Enhanced the merging of shorthand maps (`buttonShorthands`, `axisShorthands`, `povShorthands`) to avoid null pointer exceptions during deserialization.

**Testing infrastructure and coverage:**

* Added a comprehensive test suite in `ControllersTests.java` to verify correct loading and behavior of controllers from various JSON configurations, including edge cases and simulated input scenarios.
* Introduced `UnitTestingPathProvider` to resolve resource paths for test JSON files in a platform-independent way.
* Added multiple JSON resource files for testing different controller configuration scenarios, such as blank, empty list, single, multiple, and mapped controller elements. [[1]](diffhunk://#diff-c6026ed5308370a34bf1f2b5b93d9d2a03d2c08b5a8a2aa971340e0f97a12e55R1) [[2]](diffhunk://#diff-a032079597967d9051f5d5dc79783f89c1c6c91309cbf67b1c51e5942eea03cfR1-R3) [[3]](diffhunk://#diff-d1859a343457fd3c121ff2a042c1fbcf0c42b0aa4b018fe8b509b41a48b55a19R1-R19)

**Other changes:**

* Bumped the project version in `gradle.properties` to `2026.2.6`.

### CODE REVIEW

File: wpilib_interface/src/test/java/coppercore/wpilib_interface/test/UnitTestingPathProvider.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+                    + File.separator
+                    + "build"
+                    + File.separator
+                    + "resources"
+                    + File.separator
+                    + "test";
+    private String path;
+
+    public UnitTestingPathProvider() {
+        path = "";
```
[the remark this answers is not in the document]
Comment (godmar) [reply wpilib_interface/src/test/java/coppercore/wpilib_interface/test/UnitTestingPathProvider.java]: done

File: wpilib_interface/src/main/java/coppercore/wpilib_interface/controllers/Controller.java [hunk 2]
```diff
@@ -1512,6 +1489,15 @@ public Axis() {
         public double getValue() {
             return getPreparedValue();
         }
+
+        public void initialize(Controller controller) {
+            super.initialize(controller);
+            if (minValue == null)
+                minValue =
+                        (lowLevelControlElement != null) ? lowLevelControlElement.minValue : -1.0;
```
[the remark this answers is not in the document]
Comment (godmar) [reply wpilib_interface/src/main/java/coppercore/wpilib_interface/controllers/Controller.java]: I forgot about that. Keep it the way it is then. For now at least.

File: wpilib_interface/src/main/java/coppercore/wpilib_interface/controllers/Controller.java [hunk 3]
```diff
@@ -1237,12 +1237,12 @@ public static class Button extends ControlElement {
...
          *
          * <p>Inputs with absolute value less than this threshold are treated as zero and should be
-         * ignored by input processing. Default value is 0.0.
+         * ignored by input processing. Default value is 0.5.
          *
          * <p>Expected to be used with controller input values on the same scale (commonly -1.0 to
          * 1.0). Adjust as needed to filter out small/intentional noise.
          */
-        public Double threshold = 0.0;
+        public Double threshold = 0.5;
```
Comment (godmar) [inline wpilib_interface/src/main/java/coppercore/wpilib_interface/controllers/Controller.java]: Why 0.5?  This seems like a very high default.

File: wpilib_interface/src/main/java/coppercore/wpilib_interface/controllers/Controller.java [hunk 4]
```diff
@@ -1399,6 +1383,7 @@ public Button() {
          */
         public boolean isPressed() {
             double value = getPreparedValue();
+            System.out.println("Prepared Value: " + value);
```
[the remark this answers is not in the document]
Comment (avidraccoon) [reply wpilib_interface/src/main/java/coppercore/wpilib_interface/controllers/Controller.java]: @godmar Github says this is outdated and when I looked for this line in code I could not find it

File: wpilib_interface/src/main/java/coppercore/wpilib_interface/controllers/ControllerJsonRepresentation.java [hunk 5]
```diff
@@ -44,9 +45,15 @@ public Controller toJava() {
         controller.povShorthands =
                 new HashMap<String, Integer>(controller.controllerType.POVShorthands);
 
-        controller.buttonShorthands.putAll(this.buttonShorthands);
-        controller.axisShorthands.putAll(this.axisShorthands);
-        controller.povShorthands.putAll(this.povShorthands);
+        if (this.buttonShorthands != null) {
```
Comment (godmar) [inline wpilib_interface/src/main/java/coppercore/wpilib_interface/controllers/ControllerJsonRepresentation.java]: I assume that this can be `null` if the JSON representation is missing a `buttonShorthands` field.

File: wpilib_interface/src/main/java/coppercore/wpilib_interface/controllers/Controller.java (hunk 3 above)
Comment (avidraccoon) [reply wpilib_interface/src/main/java/coppercore/wpilib_interface/controllers/Controller.java]: @godmar If we want to use a trigger as a button this would be about half way through the value range and zero would lead to the button always being in the default position. So I thought this was the next best default other than zero. If you think of any that would be better I will gladly replace it.

File: wpilib_interface/src/main/java/coppercore/wpilib_interface/controllers/Controller.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (avidraccoon) [reply wpilib_interface/src/main/java/coppercore/wpilib_interface/controllers/Controller.java]: @godmar I cant find this debug statement anywhere in code, and GitHub is marking it as outdated.

File: wpilib_interface/src/main/java/coppercore/wpilib_interface/controllers/ControllerJsonRepresentation.java (hunk 5 above)
Comment (avidraccoon) [reply wpilib_interface/src/main/java/coppercore/wpilib_interface/controllers/ControllerJsonRepresentation.java]: @godmar with this code only running at startup I felt like adding this safeguard just in case was fine, but I can remove it if you want.

**Review by godmar — APPROVED** [review]
Merge and integrate if tested.

**Comment by avidraccoon** [comment]
> Could you test from maven local before you craft a new release?

I had already made a release by the time I had seen this comment and also publishToMavenLocal is not working for me
